### ISSUE TITLE
batch_size defaults are poorly chosen, training data lost

### ISSUE THREAD

**Issue by jhihn** [keras-team/keras#9492]
https://keras.io/models/model/ states several times that batch_size if not supplied, defaults to 32. This is bad. I'll even argue it is formally wrong. If someone passes in an X of more than 32, it is reasonable for the developer to expect that the model is using all data data supplied to it, rather than only the first 32 entries.  The fitting would not occur over all the data being supplied, which the developer would believe is happening, because, they supplied it. To *silently* ignore data being supplied I have to argue is an error. It doesn't even warn when supplied size is greater than 32, and  batch_size is not supplied that it will throw training data away. This kind of silent discarding of data will lead to incompletely trained models.

batch_size should always default the length of the inputs supplied, unless otherwise supplied.

**Comment by Dref360** [comment]
I think you're confusion batch_size with steps_per_epoch. In fit, the steps_per_epoch will always be `dataset_lenght / batch_size`. The batch_size is just the number of samples you feed at once to the model.

**Comment by jhihn** [comment]
I hope so, but I don' think so. From the page I linked:
```
fit
fit(self, x=None, y=None, batch_size=None, epochs=1, verbose=1, callbacks=None, validation_split=0.0, validation_data=None, shuffle=True, class_weight=None, sample_weight=None, initial_epoch=0, steps_per_epoch=None, validation_steps=None)
Trains the model for a fixed number of epochs (iterations on a dataset).

Arguments

...
batch_size: Integer or None. Number of samples per gradient update. If unspecified, batch_size will default to 32.
```

```
evaluate
evaluate(self, x=None, y=None, batch_size=None, verbose=1, sample_weight=None, steps=None)
Computes the loss on some input data, batch by batch.

Arguments
...
batch_size: Integer. If unspecified, it will default to 32.
```
```
evaluate
evaluate(self, x=None, y=None, batch_size=None, verbose=1, sample_weight=None, steps=None)
Returns the loss value & metrics values for the model in test mode.

Computation is done in batches.

Arguments

...
batch_size: Integer or None. Number of samples per evaluation step. If unspecified, batch_size will default to 32.
```

```
predict
predict(self, x, batch_size=None, verbose=0, steps=None)
Generates output predictions for the input samples.

Computation is done in batches.

Arguments
...
x: The input data, as a Numpy array (or list of Numpy arrays if the model has multiple outputs).
batch_size: Integer. If unspecified, it will default to 32.
```

**Comment by Dref360** [comment]
That's not what the docs says.
`Number of samples per gradient update` A gradient update is done after each batch. This is why it's called Stochastic GD.

`Number of samples per evaluation step` A step is done multiple time per epoch. An evaluation step is just taking a batch and feed it to the network.
`

**Comment by jhihn** [comment]
OOOH, I'm an idiot.
Thank you.
